tests/unit/test_benchmark.py

```diff
@@ -135,41 +138,57 @@ def test_initialise_solvers() -> None:
     """
     Test the :func:`initialise_solvers`.
 
-    Verify that the returned list contains callable functions that produce
+    Verify that the returned dictionary contains callable functions that produce
     valid solver instances.
     """
     # Create a mock dataset (UMAP-transformed) with arbitrary values
     mock_data = Data(jnp.array([[0.1, 0.2], [0.3, 0.4], [0.5, 0.6], [0.7, 0.8]]))
     key = random.PRNGKey(42)
-
-    solvers = initialise_solvers(mock_data, key)
-    for solver in solvers:
-        solver_instance = solver(1)  # Instantiate with a coreset size of 1
-        assert isinstance(solver_instance, (MapReduce, RandomSample, RPCholesky)), (
-            f"Unexpected solver type: {type(solver_instance)}"
-        )
-
-
-def test_get_solver_name():
-    """
-    Test `get_solver_name` function to ensure it returns correct solver names.
-    """
-    # Create a KernelHerding solver
-    herding_solver = KernelHerding(coreset_size=5, kernel=SquaredExponentialKernel())
-
-    # Wrap it in MapReduce
-    map_reduce_solver = MapReduce(base_solver=herding_solver, leaf_size=15)
-
-    assert get_solver_name(lambda _: herding_solver) == "KernelHerding", (
-        "Expected 'KernelHerding' but got something else."
-    )
-
-    assert get_solver_name(lambda _: map_reduce_solver) == "KernelHerding", (
-        "Expected 'KernelHerding' from MapReduce solver but got something else."
-    )
-
-
-@pytest.mark.parametrize("n", [10, 100, 1000])
+    g = 1
+    leaf_size = 0  # Set to 0 for simplicity in testing, adjust as needed
+
+    # Initialise solvers
+    solvers = initialise_solvers(mock_data, key, g, leaf_size)
+
+    # Ensure solvers is a dictionary with the expected keys
+    expected_solver_keys = [
+        "Random Sample",
+        "RP Cholesky",
+        "Kernel Herding",
+        "Stein Thinning",
+        "Kernel Thinning",
+        "Compress++",
+        "Probabilistic Iterative Herding",
+    ]
+    assert set(solvers.keys()) == set(expected_solver_keys), "Solver keys mismatch"
```

Comment:
```suggestion
    assert set(solvers.keys()) == set(expected_solver_keys)
```
We probably don't need the string error messages here; the error is obvious from the assert comparison, and pytest's assert rewriting means we'll have the values present to ease debugging.

Reply:
(Same comment for all other `assert predicate, message` instances!)